### PR TITLE
makeFile: Remove vendored go

### PR DESCRIPTION
**Pull request by RamLavi**
**What this PR does / why we need it**:
Since we have the go.mod "go" directive it is safe enough to ensure that the golang compiler used has the needed capabilities. This change remove the infra to use a vendored golang tool chain.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RamLavi** [comment]
/hold
seems like the lane image doesn't have go.

**Comment by qinqon** [comment]
/lgtm
/approve
we need to install go from automation or prow-jobs

File: Makefile [hunk 1]
```diff
@@ -11,34 +11,25 @@ BIN_DIR = $(CURDIR)/build/_output/bin/
...
 all: generate
 
-$(GO):
-	hack/install-go.sh $(BIN_DIR) > /dev/null
-
 # Run tests
-test: $(GO)
-	$(GO) test ./pkg/... ./cmd/... -coverprofile cover.out
+test:
+	go test ./pkg/... ./cmd/... -coverprofile cover.out
```
Comment (oshoval) [inline Makefile:29]: doesnt it means that running it locally might override the go version that the user already have ?
it isn't good if so

File: hack/install-go-gvm.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash -e
+
+install_gvm() {
```
Comment (oshoval) [inline hack/install-go-gvm.sh:3]: why gvm is better than the regular install we had ?

**Comment by RamLavi** [comment]
closing this for now. Will revisit later

**Pull request closed without merging**